starter-code/app.js

```diff
@@ -3,3 +3,78 @@ const app = express();
 const Chuck  = require('chucknorris-io');
 const client = new Chuck();
 
+// -- setup
+
+app.set('views', __dirname + '/views');
+app.set('view engine', 'ejs');
+
+
+// -- routes
+
+
+
+// our first Route:
+app.get("/random", (req, res, next) => {
+    // Retrieve a random chuck joke
+    client.getRandomJoke()
+        .then((joke) => {
+            const data = {
+                joke: joke
+            };
+            res.render("joke", data);
+        }).catch((err) => {
+        // handle error
+        });
+});
+
+
+
+// categories
+app.get("/categories", (req,res,next)=>{
+
+client.getJokeCategories()
+    .then((categories) =>{
+     const data = {categories};
+    res.render("categories",data);
+}).catch((err) => {
+    // handle error
+    console.log(err);
+    });
+}
+
+)
+
+app.get("/categories", (req,res,next)=>{
+    
+    client.getJokeCategories()
+        .then((categories) =>{
+         const data = {categories};
+        res.render("categories",data);
+    }).catch((err) => {
+        // handle error
+        console.log(err);
+        });
+    }
+    
+    )
+
+
+    app.get("/categories/:categoryId", (req, res, next) => {
+        const data = req.params.categoryId;
+        
+        client
+          .getRandomJoke(data)
+          .then(response => {
```

Comment:
Try to be consistent, before you call this kind of response joke instead of response.